### PR TITLE
Single contract

### PR DESCRIPTION
**Pull request by potomak**
This PR introduces a complete rewrite of the contract, to migrate from a multi deploy to a single deploy setup, and it uses Truffle to rationalize the contract development.

#### TODO

- [x] Missing tests
  - [x] `rollbackFunds`
  - [x] `withdrawFunds`
  - [x] `withdrawSpareFunds`

### CODE REVIEW

**Comment by potomak** [comment]
@bugant I think this PR is ready to be reviewed now. There are still a bunch of TODOs, but I wrote tests for each of the public functions of the the contract.

**Comment by potomak** [comment]
Note: time dependent tests are flaky. They fail sometimes because testrpc isn't fast enough to run transactions in less than a second, so it could happen that they fail. I should find a more reliable way to test strict time boundaries.

File: contracts/Issuehunter.sol [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+    // Only the issue manager can invoke this function.
+    //
+    // The issue manager must verify that:
+    //
+    // 1. the proposed resolution is a real solution for the selected issue, for
+    //    instance by checking that the project's master branch includes the
+    //    commit SHA that has been proposed as a resolution
+    // 2. the resolution submitter address is included in one of the commit
+    //    messages in the branch that contains the resolution
+    function verifyResolution(bytes32 issueId, address resolutor) {
```
Comment (bugant) [inline contracts/Issuehunter.sol]: Since a resolutor can submit more than one resolutions (and these will overwrite each other), I was thinking about race conditions here. Suppose a resolutor call the contract to submit a new resolution just before the issueManager call verifyResolution, what would you expect? Maybe we can also add the commitSHA here and fail if this is different from `campaigns[issueId].resolutions[resolutor]`

File: contracts/Migrations.sol [hunk 2]
```diff
@@ -0,0 +1,25 @@
+pragma solidity ^0.4.11;
+
+
+contract Migrations {
+    address public owner;
+    uint public lastCompletedMigration;
+
+    modifier restricted() {
+        require(msg.sender == owner);
+        _;
```
Comment (bugant) [inline contracts/Migrations.sol:10]: What's this?

**Review by bugant — APPROVED** [review]

File: contracts/Issuehunter.sol (hunk 1 above)
Comment (potomak) [reply contracts/Issuehunter.sol]: I think each transaction is atomic and applied independently, but you're right that this function should take both an address and a commit sha as parameters. I will update it!

File: contracts/Migrations.sol (hunk 2 above)
Comment (potomak) [reply contracts/Migrations.sol:10]: This is a _modifier_. When you use a modifier in a function, the code of the target function will be replaced with the modifier template hydrated with the target function. That's a common pattern to make modular contracts, see http://solidity.readthedocs.io/en/develop/common-patterns.html.

Comment (potomak) [reply contracts/Migrations.sol:10]: I mean it's a _special symbol in a modifier_, see http://solidity.readthedocs.io/en/develop/contracts.html#function-modifiers.

File: contracts/Issuehunter.sol (hunk 1 above)
Comment (potomak) [reply contracts/Issuehunter.sol]: @bugant I fixed `verifyResolution` to handle the special case that you suggested.
